codeworld-api/src/CodeWorld/Picture.hs

```diff
@@ -197,24 +210,23 @@ solidLoop :: HasCallStack => [Point] -> Picture
 solidLoop ps = SolidClosedCurve callStack ps
 
 {-# WARNING solidLoop "Please use solidClosedCurve instead of solidLoop." #-}
+rectangleToPolygon :: CallStack -> Double -> Double -> Picture
+rectangleToPolygon cs w h = Polygon cs [(-w / 2, -h / 2), (w / 2, -h / 2), (w / 2, h / 2), (-w / 2, h / 2)]
 
 -- | A thin rectangle, with this width and height
 rectangle :: HasCallStack => Double -> Double -> Picture
-rectangle w h =
-    polygon [(-w / 2, -h / 2), (w / 2, -h / 2), (w / 2, h / 2), (-w / 2, h / 2)]
+rectangle w h = 
```

Comment:
All of these user-facing utilities like `rectangle`/`solidRectangle`/`thickRectangle` should follow the exact same pattern, which is:

```haskell
lowerCaseName = UpperCaseName callStack
```

In other words:

```haskell
rectangle :: CallStack -> Double -> Double -> Picture
rectangle = Rectangle callStack

solidRectangle :: HasCallStack => Double -> Double -> Picture
solidRectangle = SolidRectangle callStack

thickRectangle :: HasCallStack => Double -> Double -> Double -> Picture
thickRectangle = ThickRectangle callStack
```

Just stick to that pattern.  If you think you need to do something different, that's a good time to ask for help

Reply:
Aye aye Cap'n. I'll work on it tomorrow :D

Reply:
so I think I'm not sure about how to use the helper function and make a new constructor. I can get two of the functions to type check, but I'm still using the helper function, which uses Polygon :/